### PR TITLE
Refactoring, rgkit integration, and more features!

### PR DESCRIPTION
**Pull request by bwnance**
Refactoring from MB6

### CODE REVIEW

**Comment by bwnance** [comment]
This now also includes rgkit integration.

**Comment by bwnance** [comment]
Now allows for changing the number of matches and the precision via the command line
(-pr and -m) as mentioned in issue https://github.com/Spferical/rgtuner/issues/4

File: rgtuner.py [hunk 1]
```diff
@@ -107,35 +97,26 @@ def optimize_variable(enemies, variable, robot_file, processes):
...
-                    return [int(scores[0]),int(scores[1]), 0,'tie']
-
-
-                
-    except KeyboardInterrupt:
-        p.terminate()
-
-
-def versus(bot1, bot2, processes):
+    #what is this???
```
Comment (Spferical) [inline rgtuner.py]: What is this?

**Comment by bwnance** [comment]
@Spferical It's rgkit integration, instead of spawning processes to run matches through the console, it does so through rgkit's runner class

**Comment by Spferical** [comment]
Sorry I meant the comment, "#what is this???" It took me a bit to realize it was emphasis for the comment 3 lines below.

**Comment by bwnance** [comment]
Ah, haha

**Comment by Spferical** [comment]
So, awesome! The changes are great! I have just a few issues.

> things are imported where needed if where they are needed is only one function (imported to local scope, local lookups are faster)

There is a huge overhead for importing multiple times. Python caches imports, but the time taken for the cache lookup outweighs the speed boost from local lookups.
See https://gist.github.com/Spferical/c41bc85c4f1240c1a56f for a demonstration of this holding even when library functions are called a lot.
Also, this is a micro-optimization, discouraged by style guides like PEP 8, and not really warranted by anything. There are much better ways to optimize the code.

Which brings me to the rgkit integration. This is great, because it means PyPy can be used to run the code, offering huge speed improvements. It runs rgtuner about twice as fast as regular CPython for me!

Also, you can remove the commented-out lines of code. Keeping those around is what version control is for.

Trailing whitespace removal, formatting, range -> xrange, precision option, are all great! I'd love to merge this as soon as the imports and commented-out-lines are changed.

**Comment by bwnance** [comment]
Tada!

**Comment by Spferical** [comment]
Huzzah!

**Comment by MB6** [comment]
I recomend not using with pypy as my tests show memory usage is at least (inexplicably) 10x more (20mb and stable vs 200 mb and rising)

**Comment by Spferical** [comment]
That's interesting. Maybe it has something to do with robot code being repeatedly loaded into rgkit. Garbage collection differences?

I just tested and got similar results. Plus, with six processes, the memory usage can quickly climb into the gigabytes.

I personally won't stop using PyPy while I still have enough memory to contain it since it's way faster, but, dang, that is something to keep in mind.

**Pull request merged**